sse2neon.h

```diff
@@ -1469,11 +1478,33 @@ FORCE_INLINE __m128 _mm_cvtpi8_ps(__m64 a)
 // will generate 0x7FFF, rather than 0x8000, for input values between 0x7FFF and
 // 0x7FFFFFFF.
 //
+//   FOR j := 0 to 3
+//     i := 16*j
+//     k := 32*j
+//     IF a[k+31:k] >= FP32(0x7FFF) && a[k+31:k] <= FP32(0x7FFFFFFF)
+//       dst[i+15:i] := 0x7FFF
+//     ELSE
+//       dst[i+15:i] := Convert_FP32_To_Int16(a[k+31:k])
+//     FI
+//   ENDFOR
+//
 // https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_cvtps_pi16
 FORCE_INLINE __m64 _mm_cvtps_pi16(__m128 a)
 {
-    return vreinterpret_m64_s16(
-        vmovn_s32(vreinterpretq_s32_m128i(_mm_cvtps_epi32(a))));
+    __m128 i16Min = _mm_set_ps1(INT16_MIN);
```

Comment:
I have no idea about the necessity of these changes.

Reply:
The original implementation does not take care of the description
```
Note: this intrinsic will generate 0x7FFF,
rather than 0x8000, for input values between 0x7FFF and 0x7FFFFFFF.
```

The intrinsic should set the corresponding value
```
0x7FFF <= input <= 0x7FFFFFFF
output = 0x7FFF(INT16_MAX)
0x8000 < input < 0x7FFF
output = _mm_cvtps_epi32(input)
other situation
output = 0x8000(INT16_MIN)
```

The original implementation calls the conversion intrinsic `_mm_cvtps_epi32()` only.

Reply:
Mention the above in git commit message.

Reply:
Done.